llvm/test/Transforms/InstCombine/vec_shuffle-phi-multiuse.ll

```diff
@@ -0,0 +1,57 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --version 5
+; RUN: opt < %s -S -passes=instcombine | FileCheck %s
+
+define void @f(ptr %p_begin, ptr %p_end, ptr %out) {
```

Comment:
Is it worth adding tests for cases where it should not simplify (e.g. too many users, some users can't simplify)?

Reply:
added more tests